plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/StandardColumnMappings.java

```diff
@@ -381,8 +381,7 @@ private static LocalTime toLocalTime(Time sqlTime)
     {
         // Time.toLocalTime() does not preserve second fraction
         return sqlTime.toLocalTime()
-                // TODO is the conversion correct if sqlTime.getTime() < 0?
-                .withNano(toIntExact(MILLISECONDS.toNanos(sqlTime.getTime() % 1000)));
+                .withNano(toIntExact(MILLISECONDS.toNanos(Math.floorMod(sqlTime.getTime(), 1000L))));
```

Comment:
static import `floorMod` (is it already imported here?)

Reply:
Yes, good catch. Fixed.